### PR TITLE
OCD-4653 - Create Developer widgets

### PR DESCRIPTION
**Pull request by tmy1313**

### CODE REVIEW

File: src/app/navigation/navigation-top.html [hunk 1]
```diff
@@ -89,6 +89,7 @@
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc'])"><a ui-sref="administration.api-keys"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.api-keys')"></i> API Keys</a></li>
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc', 'chpl-onc-acb'])"><a ui-sref="administration.change-requests"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.change-requests')"></i> Change Requests</a></li>
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc'])"><a ui-sref="administration.system-maintenance"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.system-maintenance')"></i> System Maintenance</a></li>
+              <li ng-if="vm.hasAnyRole(['chpl-admin'])"><a ui-sref="power-bi"><i class="fa fa-caret-right" ng-if="vm.isActive('sys-admin.power-bi')"></i>Power BI Report</a></li>
```
Comment (kekey1) [inline src/app/navigation/navigation-top.html]: This is just a nitpicky thing but I don't like this being called "Power BI" anything. I would like "CHPL By the Numbers" or "Statistics Dashboard" or just "Dashboard" or something. It seems secondary (from a UI point of view) that we are using Power BI to display this dashboard.

**Review by kekey1 — DISMISSED** [review]

File: src/app/navigation/navigation-top.html (hunk 1 above)
Comment (andlar) [reply src/app/navigation/navigation-top.html]: I like this idea. We don't know what we're actually going to be calling this thing, but why not try to guess. Even just "chplDashboard" or something would match our patterns and could be what we eventually use

**Review by kekey1 — DISMISSED** [review]

File: src/app/api/reports.jsx [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { useQuery } from 'react-query';
+
+import { useAxios } from './axios';
+
+const useFetchReportUrl = (reportName) => {
+  const axios = useAxios();
+  return useQuery(['report-url', reportName], async () => {
+    if (reportName) {
+      const response = await axios.get(`report-data/${reportName}/url`);
+      console.log('reports.jsx');
```
Comment (andlar) [inline src/app/api/reports.jsx]: `report-url` and `report-data` should match. I don't know if I care which one is used, but they should be the same

File: src/app/api/reports.jsx [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+      console.log(response.data);
```
Comment (andlar) [inline src/app/api/reports.jsx]: Should remove debug lines

File: src/app/navigation/navigation-top.html [hunk 4]
```diff
@@ -89,6 +89,7 @@
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc'])"><a ui-sref="administration.api-keys"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.api-keys')"></i> API Keys</a></li>
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc', 'chpl-onc-acb'])"><a ui-sref="administration.change-requests"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.change-requests')"></i> Change Requests</a></li>
               <li ng-if="vm.hasAnyRole(['chpl-admin', 'chpl-onc'])"><a ui-sref="administration.system-maintenance"><i class="fa fa-caret-right" ng-if="vm.isActive('administration.system-maintenance')"></i> System Maintenance</a></li>
+              <li ng-if="vm.hasAnyRole(['chpl-admin'])"><a ui-sref="dashboard"><i class="fa fa-caret-right" ng-if="vm.isActive('sys-admin.dashboard')"></i>Dashboard</a></li>
```
Comment (andlar) [inline src/app/navigation/navigation-top.html:92]: Issue #1 on the ticket says ONC should have access to these reports, and the ticket description says that as well

File: src/app/pages/resources/dashboard/dashboard-wrapper.jsx [hunk 5]
```diff
@@ -0,0 +1,18 @@
+import React from 'react';
+
+import Dashboard from './dashboard';
+
+import AppWrapper from 'app-wrapper';
+
+function DashboardWrapper() {
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard-wrapper.jsx]:
```suggestion
function ChplDashboardWrapper() {
```

Our naming convention is that our components all start with Chpl

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 6]
```diff
@@ -0,0 +1,155 @@
...
+} from '@material-ui/core';
+import { useFetchReportUrl } from 'api/reports';
+import { palette } from 'themes';
+
+import AssessmentOutlinedIcon from '@material-ui/icons/AssessmentOutlined';
+import DashboardOutlinedIcon from '@material-ui/icons/DashboardOutlined';
+import GroupOutlinedIcon from '@material-ui/icons/GroupOutlined';
+import HighlightOutlinedIcon from '@material-ui/icons/HighlightOutlined';
+
+const useStyles = makeStyles((theme) => ({
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx]:
```suggestion
const useStyles = makeStyles({
```

Passing a function into the makeStyles was what we did early on, but is not the standard going forward

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 7]
```diff
@@ -0,0 +1,155 @@
...
+  container: {
+    height: '1200px',
+    padding: theme.spacing(8),
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx:22]: The `theme` variable is not imported anywhere in this file, so this value can't exist. It should be imported in line 12 with the `import { palette, theme }` line

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 8]
```diff
@@ -0,0 +1,155 @@
...
+  Button,
+  makeStyles,
+} from '@material-ui/core';
+import { useFetchReportUrl } from 'api/reports';
+import { palette } from 'themes';
+
+import AssessmentOutlinedIcon from '@material-ui/icons/AssessmentOutlined';
+import DashboardOutlinedIcon from '@material-ui/icons/DashboardOutlined';
+import GroupOutlinedIcon from '@material-ui/icons/GroupOutlined';
+import HighlightOutlinedIcon from '@material-ui/icons/HighlightOutlined';
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx:14]: These are external imports, and should be part of the first set of imports, immediately after the line with `@material-ui/core`

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 9]
```diff
@@ -0,0 +1,155 @@
...
+  Box,
+  Card,
+  CardContent,
+  Typography,
+  Container,
+  Button,
+  makeStyles,
+} from '@material-ui/core';
+import { useFetchReportUrl } from 'api/reports';
+import { palette } from 'themes';
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx]: These are internal imports (from our project) and should be separated from the external imports by a new line. I _think_ our js-linter should report on that as an issue, but I don't know if you have it set up

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 10]
```diff
@@ -0,0 +1,155 @@
...
+                  <CardContent>
+                    <Typography gutterBottom variant="h6">
+                      <b>Welcome to CHPL By The Numbers Dashboard</b>
+                    </Typography>
+                    <Typography gutterBottom>
+                      A dynamic reporting suite powered by PowerBI, providing detailed insights and analytics derived from CHPL data. This tool offers interactive reports with robust click-through capabilities, allowing users to explore and analyze data seamlessly. Each report is designed to be user-friendly, enabling in-depth exploration of key metrics and trends, with the flexibility to dive deeper into the numbers that matter most.
+                    </Typography>
+                    <Box mt={8} mb={4} display="flex" flexDirection="row" flexWrap="wrap" gridGap={32}>
+                      {[
+                        { text: 'Developer Reporting', page: 'developerReport' },
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx]: It feels weird having two arrays of the same data defined in different places. Can this set of data be included in the array above that has the icons, and then this `.map` can just be something like `pages.map((item) =>`?

File: src/app/pages/resources/dashboard/dashboard.jsx [hunk 11]
```diff
@@ -0,0 +1,155 @@
...
+                      width="100%"
+                      height="900px"
+                      src={developerStatisticsReportUrl?.data?.reportUrl}
+                      frameBorder="0"
+                      allowFullScreen
+                    />
+                  </CardContent>
+                </Card>
+              )}
+              {/* Add more conditionally rendered cards here based on `activeCard` value */}
```
Comment (andlar) [inline src/app/pages/resources/dashboard/dashboard.jsx:141]: Do we have a menu with four icons on it, but only one of them works?

File: src/app/pages/resources/dashboard/index.js [hunk 12]
```diff
@@ -0,0 +1,3 @@
+import Dashboard from './dashboard';
```
Comment (andlar) [inline src/app/pages/resources/dashboard/index.js]: We started with using these index.js files, but they just get in the way and don't provide any value. Even in this code you're not actually using this file anywhere

**Review by andlar — CHANGES_REQUESTED** [review]
I noticed the PR is not up to date with staging. Not sure how much that'll change things, but it'll need to be updated before it can be merged, at least

**Review by andlar — DISMISSED** [review]
Leaving some comments open, as they're likely going to be dependent on PO input. Would be okay moving forward in the meantime

**Review by kekey1 — DISMISSED** [review]

**Review by andlar — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — DISMISSED** [review]

**Review by kekey1 — APPROVED** [review]

**Review by andlar — APPROVED** [review]

**Pull request merged**